Products/Zuul/facades/metricfacade.py

```diff
@@ -51,24 +53,144 @@ def _isRunningFromUI(context):
+                                              auth=auth,
+                                              timeout=timeout)
+        except requests.exceptions.Timeout as e:
+            raise ServiceConnectionError(
+                'Timed out waiting for response from metric service: %s' % e, e)
+        status_code = response.status_code
+        if not (status_code >= 200 and status_code <= 299):
+            log.error('Server response error: %s %s', status_code, response)
+            raise ServiceResponseError(response.reason, status_code, request,
+                                       response, response.content)
+
+        content = response.json()
+
+        return content
+
+    def _request(self, path, request, timeout):
+        auth = None
+        if not self._req_session.cookies.get(Z_AUTH_TOKEN):
+            if self._credentials:
+                auth = self._credentials
+            else:
+                auth = self._global_credentials
+
+        try:
+            return self._server_request(path, request, auth, timeout)
+        except ServiceResponseError as e:
+            if e.status == 401 and auth != self._global_credentials:
+                # Try using global credentials.
+                auth = self._global_credentials
+
+                if Z_AUTH_TOKEN in self._req_session.cookies:
+                    del self._req_session.cookies[Z_AUTH_TOKEN]
+
+                log.info('Authorization failed. Trying to use global credentials')
+                return self._server_request(path, request, auth, timeout)
+            else:
+                raise
+
+    def request(self, path, request, timeout=10):
+        """
+        Performs request to Metrics Server.
+
+        If `auth_token` is not None, it will be used for authentication.
+        Otherwise, check if `credentials` is not None and uses them. And
+        finally will use `global_credentials` if both of previous ones are None.
+
+        In case of authentication error, tries to use global credentials for
+        authentication.
+
+        :param path: path to API method on Metric Server without scheme, host
+                     and port. (i.e. /api/performance/query)
+        :param request: dict with request parameters
+        :param timeout: timeout in seconds to wait for response from server
+        :return: decoded response from server or None if error occurred
+        """
+        try:
+            return self._request(path, request, timeout)
+        except ServiceResponseError as e:
+            # there was an error returned by the metric service, log it here
+            log.error('Error fetching request: %s \n'
+                      'Response from the server (return code %s): %s',
+                      request, e.status, e.content)
+            return None
+        except ServiceConnectionError as e:
+            log.error('Error connecting with request: %s \n%s', request, e)
+            return None
```

Comment:
The `return None` lines in the exception handlers are redundant here.  The default behavior when a function "falls off" the end is to return `None`.